### PR TITLE
PGVector#update_texts

### PR DESCRIPTION
**Pull request by dcluna**
This commit implements (tentative) pgvector support for upsert functionality, which as far as I can understand is all that is required to make it work correctly with the Rails integration.

This is open to discussion and modification - please let me know if there's anything to improve here.

### CODE REVIEW

**Comment by andreibondarev** [comment]
@dcluna Please run `standardrb --fix` and commit. Thank you for your contribution here!

File: spec/langchain/vectorsearch/pgvector_spec.rb [hunk 1]
```diff
@@ -47,7 +47,44 @@
       it "adds texts" do
         command = subject.add_texts(texts: ["Hello World", "Hello World"])
```
Comment (dcluna) [inline spec/langchain/vectorsearch/pgvector_spec.rb:48]: The way this was written _suggests_ that ids are optional in this method - however, from the usage I could find (for instance, https://github.com/andreibondarev/langchainrb/blob/7d409a378ab300f4641a71b74d25e5cd9593122f/lib/langchain/active_record/hooks.rb#L52-L55) it seems like this is not the case.

I'd be more than happy to enforce the existence of texts/ids in tandem, but not sure if there is an (upcoming?) use case that relies on this API.

**Review by dcluna — COMMENTED** [review]
@andreibondarev Please take a look at this comment - had to write a corner case for when ids are not required in `add_texts`, but if this is not necessary I'd prefer to have the original API using the same method for both add/update operations.

**Comment by dcluna** [comment]
@andreibondarev Also, not sure if CI did not trigger due to me force-pushing the branch or if it's actually manually triggered by you - in any case, could you please make it run so this PR can be merged?

**Comment by andreibondarev** [comment]
@dcluna No, at least at `add_texts()` the IDs are not required because if you don't pass IDs they should be created automatically.

**Pull request merged**